core/pipeline/label.go

```diff
@@ -168,18 +188,27 @@ func LoadGitLabels(workdir string) ([]Label, error) {
 		},
 	)
 
-	if head.Name().IsTag() {
-		labels = append(labels, Label{
-			Name:  "dagger.io/git.tag",
-			Value: head.Name().Short(),
-		})
-	}
-
-	if head.Name().IsBranch() {
-		labels = append(labels, Label{
-			Name:  "dagger.io/git.branch",
-			Value: head.Name().Short(),
-		})
+	// check if ref is a tag or branch
+	refs, _ := repo.References()
+	err = refs.ForEach(func(ref *plumbing.Reference) error {
+		if ref.Hash() == commit.Hash {
+			if ref.Name().IsTag() {
+				labels = append(labels, Label{
+					Name:  "dagger.io/git.tag",
+					Value: ref.Name().Short(),
+				})
+			}
+			if ref.Name().IsBranch() {
+				labels = append(labels, Label{
+					Name:  "dagger.io/git.branch",
+					Value: ref.Name().Short(),
+				})
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
```

Comment:
just to be sure, is this change required because before we were incorrectly collecting the tag and the branch in PR's?

Reply:
it could work as in the past, however, the tag is not applied to the added merge commit from GitHub but to a previous commit, so we would miss it

Reply:
> just to be sure, is this change required because before we were incorrectly collecting the tag and the branch in PR's?

You are right. In the context of a PR, we would 100 % miss the tag.